onnx/large_helper.py

```diff
@@ -0,0 +1,324 @@
+        the function returns this modified copy.
+
+        Arguments:
+            file_path: model file
+            all_tensors_to_one_file: all tensors in one file
+
+        Returns:
+            modified main model proto
+        """
+        _unique_names: set[str] = set()
+
+        def _clean_name(prefix: str, name: str) -> str:
+            for c in ":/\\;,!":
+                name = name.replace(c, "")
+            base_name = name
+            i = 0
+            while name in _unique_names:
+                i += 1
+                name = f"{base_name}_{i}"
+            _unique_names.add(name)
+            return name
+
+        folder = os.path.dirname(file_path)
+        if not os.path.exists(folder):
+            raise FileNotFoundError(f"Folder {folder!r} does not exist.")
+        proto = self.model_proto.SerializeToString()
+        copy = onnx.ModelProto()
+        copy.ParseFromString(proto)
+        prefix = os.path.splitext(os.path.split(file_path)[-1])[0]
+
+        if all_tensors_to_one_file:
+            file_weight = f"{os.path.split(file_path)[1]}.weight"
+            full_file_weight = f"{file_path}.weight"
+            offset = 0
+            with open(full_file_weight, "wb") as f:
+                pass
+
+        for tensor in ext_data._get_all_tensors(copy):
+            if not ext_data.uses_external_data(tensor):
+                continue
+            prop: onnx.StringStringEntryProto | None = None
+            for ext in tensor.external_data:  # type: ignore[assignment]
+                if ext.key == "location":  # type: ignore[attr-defined]
+                    prop = ext  # type: ignore[assignment]
+            if prop is None:
+                raise RuntimeError(
+                    f"No location found for tensor name {tensor.name!r}."
+                )
+            if prop.value not in self.large_initializers:
+                raise RuntimeError(
+                    f"Unable to find large tensor named {tensor.name!r} "
+                    f"with location {prop.value!r} in "
+                    f"{sorted(self.large_initializers)}."
+                )
+            np_tensor = self.large_initializers[prop.value]
+
+            if all_tensors_to_one_file:
+                buffer = np_tensor.tobytes()
+                _set_external_data(
+                    tensor, location=file_weight, offset=offset, length=len(buffer)
+                )
+                offset += len(buffer)
+                with open(full_file_weight, "ab") as f:
+                    f.write(buffer)
+            else:
+                name = f"{_clean_name(prefix, prop.value)}.weight"
+                _set_external_data(tensor, location=name)
+                full_name = os.path.join(folder, name)
+                prop.value = name
+                with open(full_name, "wb") as f:
+                    f.write(np_tensor.tobytes())
+
+        with open(file_path, "wb") as f:
+            f.write(copy.SerializeToString())
+        return copy
```

Comment:
"copy" collides with a built in name

Reply:
copy is a module, is a build-in function?